conformance/tdl/if_some.ion

```diff
@@ -0,0 +1,132 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+(ion_1_1 "if_some special form cannot be"
+         (each "invoked as an unqualified e-expression"
+               (text "(:if_some 1 2 3)")
+               "invoked as a qualified e-expression"
+               (text "(:$ion::if_some 1 2 3)")
+               "exported"
+               (mactab (export if_some))
+               (mactab (export $ion::if_some))
+               (signals "no such macro")))
+
+(ion_1_1 "when invoked in TDL,"
+         // TODO: Can be invoked qualified and unqualified
+         //       Can be shadowed
+         (each "each argument can be elided because they are optional trailing parameters"
+               (mactab (macro foo () (.$ion::if_some)))
+               (mactab (macro foo () (.$ion::if_some (..))))
+               (mactab (macro foo () (.$ion::if_some (..) 3)))
+               (then (toplevel 0 ('$#:foo') 1)
+                     (produces 0 1))))
+
+(ion_1_1 "when the first argument is"
+         (each "an empty expression group"
+               (mactab (macro is_some () (.$ion::if_some (..) true false)))
+               "a macro that produces nothing"
+               (mactab (macro is_some () (.$ion::if_some (.$ion::none) true false)))
+               "an expression group with one macro that produces nothing"
+               (mactab (macro is_some () (.$ion::if_some (..(.$ion::none)) true false)))
+               "an expression group with multiple macros that produce nothing"
+               (mactab (macro is_some () (.$ion::if_some (..(.$ion::none)(.$ion::none)) true false)))
+               (then "then if_some should produce the false branch"
+                     (toplevel ('#$:is_some'))
+                     (produces false)))
+         (each "a single value"
```

Comment:
Should this be "some" or "not none" rather than "a single value", given that it describes cases where the first argument is more than a single value?

Reply:
No. the `each` clause allows you to intersperse descriptions with fragments, so this particular description applies only to the fragments before the next description. In this case, it only applies to `(mactab (macro is_some () (.$ion::if_some 0 true false)))`.